### PR TITLE
ui: Treat thread fallback the same as a regular reply

### PR DESCRIPTION
**Pull request by jplatte**
… for now, until we add proper support for threads.

Closes https://github.com/vector-im/element-x-ios/issues/477.

### CODE REVIEW

File: crates/matrix-sdk-ui/src/timeline/event_item/content.rs [hunk 1]
```diff
@@ -272,6 +264,22 @@ pub struct InReplyToDetails {
...
 
+impl InReplyToDetails {
+    pub fn new(
+        event_id: OwnedEventId,
+        timeline_items: &Vector<Arc<TimelineItem>>,
+    ) -> InReplyToDetails {
+        let event = timeline_items
+            .iter()
+            .filter_map(|it| it.as_event())
+            .find(|it| it.event_id() == Some(&*event_id))
```
Comment (bnjbvr) [inline crates/matrix-sdk-ui/src/timeline/event_item/content.rs:275]: Can this be folded into a single `find_map`?

**Review by bnjbvr — APPROVED** [review]
lgtm, thanks; if it's easy to add a test for that, it would be awesome to have it!

File: crates/matrix-sdk-ui/src/timeline/event_item/content.rs (hunk 1 above)
Comment (jplatte) [reply crates/matrix-sdk-ui/src/timeline/event_item/content.rs:275]: Sure.

Comment (jplatte) [reply crates/matrix-sdk-ui/src/timeline/event_item/content.rs:275]: Though honestly, I don't find it easier to read:

```rust
.find_map(|it| it.as_event().filter(|evt| evt.event_id() == Some(&*event_id)))
```

Comment (bnjbvr) [reply crates/matrix-sdk-ui/src/timeline/event_item/content.rs:275]: Uh indeed, it's either `filter()` or `and_then()` to extract the `event_id()`, nevermind.

**Pull request merged**